packages/main/src/plugin/kubernetes/contexts-manager-experimental.ts

```diff
@@ -61,29 +68,77 @@ export class ContextsManagerExperimental {
   }
 
   private async onAdd(event: DispatcherEvent): Promise<void> {
-    const previous = this.#healthCheckers.get(event.contextName);
-    previous?.dispose();
-    const newHC = new ContextHealthChecker(event.config);
-    newHC.onStateChange(this.onStateChange.bind(this));
-    this.#healthCheckers.set(event.contextName, newHC);
-    await newHC.start({ timeout: HEALTH_CHECK_TIMEOUT_MS });
+    // register and start health checker
+    const previousHealthChecker = this.#healthCheckers.get(event.contextName);
+    previousHealthChecker?.dispose();
+    const newHealthChecker = new ContextHealthChecker(event.config);
+    this.#healthCheckers.set(event.contextName, newHealthChecker);
+    newHealthChecker.onStateChange(this.onStateChange.bind(this));
+
+    newHealthChecker.onReachable(async (state: ContextHealthState) => {
+      // register and start permissions checker
+      const previousPermissionsChecker = this.#permissionsCheckers.get(state.contextName);
+      previousPermissionsChecker?.dispose();
+
+      const namespace = state.kubeConfig.getNamespace();
+      const newPermissionChecker = new ContextPermissionsChecker(state.kubeConfig, {
+        attrs: {
+          namespace,
+          group: '*',
+          resource: '*',
+          verb: 'watch',
+        },
+        resources: ['pods', 'deployments'],
```

Comment:
Why are we only checking the permissions for pods and deployments ?

Reply:
One step at a time. For now, these are the resources needed to provide information on contexts